src/utils.js

```diff
@@ -1,241 +1,227 @@
-		c = str[i];
-		if (c.match(spaceRe) && !insideAString) {
-			flushBuffer();
-			continue;
-		}
-		if (c.match(doubleQuoteRe)) {
-			if (insideAString) {
-				flushBuffer();
-			}
-			insideAString = !insideAString;
-			continue;
-		}
-		buff += c;
-	}
-	flushBuffer();
-	return result;
-};
-
-/**
-object clone (object)
-
-Clone an object.
-*/
-exports.clone = function(obj) {
-	var copy, key;
-	if (obj === null || typeof obj !== "object") {
-		return obj;
-	}
-	copy = obj.constructor();
-	for (key in obj) {
-		if (!obj.hasOwnProperty(key)) {
-			continue;
-		}
-		copy[key] = exports.clone(obj[key]);
-	}
-	return copy;
-};
-
-/**
-int nIndexOf (string, string match, int index)
-
-Finds a match in a string at a given index
-
-Usage:
-string = "My name is Rive"
-match = " "
-index = 2
-return = 7
-
-Summary: It will look for a second space in the string
-*/
-exports.nIndexOf = function(string, match, index) {
-	return string.split(match, index).join(match).length;
-};
+ * Object clone (object)
+ *
+ * Clone an object.
+ */
+
+const clone = (obj) => {
+    if (obj === null || typeof obj !== "object") {
+      return obj
+    }
+    const copy = obj.constructor()
+    for (const key in obj) {
+      if (!obj.hasOwnProperty(key)) {
+        copy[key] = clone(obj[key])
+      }
+    }
+    return copy
+  },
+
+  /**
+   * Void extend (object a, object b)
+   *
+   * Combine the properties of both objects into one. The properties from
+   * object 'b' are inserted into 'a'.
+   */
+
+  extend = (a, b) => (() => {
+    const result = []
+    for (const attr in b) {
+      const value = b[attr]
+      if (!b.hasOwnProperty.call(attr)) {
+        result.push(a[attr] = value)
+      }
+    }
+    return result
+  })(),
+
+  /**
+   * Boolean isAPromise (object)
+   *
+   * Determines if obj looks like a promise
+   */
+
+  isAPromise = (obj) => obj &&
+  obj.then &&
+  obj.catch &&
+  obj.finally &&
+  typeof obj.then === "function" &&
+  typeof obj.catch === "function" &&
+  typeof obj.finally === "function",
+
+  /**
+   * Bool isAtomic (string trigger)
+   *
+   * Determine whether a trigger is atomic.
+   */
+
+  isAtomic = (trigger) => {
+    /**
+     * Atomic triggers don't contain any wildcards or parenthesis or
+     * anything of the sort. We don't need to test the full character set
+     * just left brackets will do.
+     */
+    for (const special of[
+        "*",
+        "#",
+        "_",
+        "(",
+        "[",
+        "<",
+        "@"
+      ]) {
+      if (trigger.includes(special)) {
+        return false
+      }
+    }
+    return true
+  },
+
+  /**
+   * Int nIndexOf (string, string match, int index)
+   *
+   * Finds a match in a string at a given index
+   *
+   * Usage:
+   * string = "My name is Rive"
+   * match = " "
+   * index = 2
+   * return = 7
+   *
+   * Summary: It will look for a second space in the string
+   */
+
+  nIndexOf = (string, match, index) => string.
+split(match, index).
+join(match).length,
```

Comment:
ESLint thinks this is the way, looks strange I agree, solution is to put a // eslint-disable-line and indent normally.

Reply:
I'm not sure which style guide you're using  with eslint, but I would recommend the airbnb (eslont-config-airbnb) one since that's a very standard (I think the _most_ popular) one. For implicit returns spanning multiple lines, the returned expression should be wrapped in parentheses: https://github.com/airbnb/javascript/blob/master/README.md#arrows--paren-wrap

Reply:
I am using all the rules, I am not joking. Haha. When I can't fix something I just use // eslint-disable-line